indexer/src/database.rs

```diff
@@ -165,7 +170,21 @@ impl Database {
             })
             .collect();
 
-        let query_text = self.upsert_query(table, &self.schema[table], inserts, updates);
+        let columns = match include_id {
+            true => self.schema[table].clone(),
+            false => self.schema[table]
+                .iter()
+                .filter_map(|col| {
+                    if col == "id" {
+                        None
+                    } else {
+                        Some(col.to_owned())
+                    }
+                })
+                .collect(),
```

Comment:
Should be possible to simplify this branch a little to something like: 
```rust
            false => self.schema[table]
                .iter()
                .filter(|&col| col != "id")
                .cloned()
                .collect(),
```
I've leave the choice on whether it's acceptable to use `include_id` like this up to @tjsharp1.